packages/blade/src/components/Toast/_decisions/decisions.md

```diff
@@ -0,0 +1,251 @@
+   */
+  content: string | React.ReactNode;
+
+  /**
+   * Can be used to render an icon
+   */
+  leading?: IconComponent;
+
+  /**
+   * If true, the toast will be dismissed after few seconds
+   * 
+   * Duration for promotional toast is 6s
+   * Duration for informational toast is 4s
+   * 
+   * @default false
+   */
+  autoDismiss?: boolean;
+
+  /**
+   * Called when the toast is dismissed or duration runs out
+   */
+  onDismissButtonClick?: () => void;
+
+  /**
+   * Primary action of toast
+   */
+  action?: {
+    text: string;
+    onClick?: () => void;
+    isLoading? boolean;
+  }
+
+  /**
+   * Forwarded to react-hot-toast
+   * 
+   * This can be used to programatically update toasts by providing an id
+   */
+  id?: string; 
+}
+```
+
+### useToast
+
+The useToast hook will return few modified methods from [react-hot-toast](https://react-hot-toast.com/docs/toast): 
+
+```ts
+type useToastReturnType = {
+  /**
+   * @returns id of the toast
+   */
+  showToast: (toast: Toast) => string;
+
+  /**
+   * id of the toast to be dismissed
+   * 
+   * if id is not provided, all the toasts will be dismissed
+   */
+  dismissToast: (id?: string) => void;
+}
+```
+
+## Examples
+
+#### Removing a toast
+
+react-hot-toast provides this functionality, for more info see [react-hot-toast docs](https://react-hot-toast.com/docs/toast#dismiss-toast-programmatically)
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const Example = () => {
+  const toastId = React.useRef(null);
+  const { showToast, dismissToast } = useToast();
+
+  return (
+    <Box>
+      <Button
+        onClick={() => {
+          toastId.current = showToast({
+            color: 'success',
+            content: 'Payment Successful',
+          });
+        }}
+      >
+        Show Toast
+      </Button>
+      <Button onClick={() => dismissToast(toastId.current)}>Dismiss Toast</Button>
+    </Box>
+  );
+};
+```
+
+### Promotional Toast
+
+```jsx
+import { BladeProvider, ToastContainer, useToast } from "@razorpay/blade/components"
+
+const Example = () => {
+  const { showToast } = useToast();
+
+  return (
+    <Button
+      onClick={() => {
+        showToast({
+          type: 'promotional',
+          content: (
+            <Box>
+              <Heading>Payment Successful</Heading>
+              <Text>Amount: ₹100</Text>
+            </Box>
+          ),
+          leading: <DollarIcon />,
+          action: {
+            text: 'Okay'
+          }
+        });
+      }}
+    >
+      Do payment
+    </Button>
+  );
+};
+```
+
+<img src="./example-promotional-toast.png" width="380" alt="Promotional toast example" />
+
+
+## Motion
+
+- TBD
+
+## Accessibility
+
+- Toast components will follow the WAI-ARIA guidelines for [alert](https://www.w3.org/WAI/ARIA/apg/patterns/alert/examples/alert/). For error toast we will use the `alert` role and for informational toast we will use the `status` role.
```

Comment:
can you mention the exact attributes? also the alert role will be applied on toast or the parent container can you mention that too?

Reply:
Basically this will be same a11y as our Alert component. Just a role prop. 

And anyways the react-hot-toast will handle the accessibility props out of the box and we will just spread the props.